src/app/clusters/electrical-energy-measurement-server/ElectricalEnergyMeasurementCluster.h

```diff
@@ -17,46 +17,100 @@
+    using OptionalAttributesSet = OptionalAttributeSet<                        //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyImported::Id, //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyExported::Id, //
+        ElectricalEnergyMeasurement::Attributes::PeriodicEnergyImported::Id,   //
+        ElectricalEnergyMeasurement::Attributes::PeriodicEnergyExported::Id,   //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyReset::Id     //
+        >;
 
-    CHIP_ERROR Init();
-    void Shutdown();
+    struct Config
+    {
+        constexpr Config(EndpointId endpoint, BitMask<ElectricalEnergyMeasurement::Feature> featureFlags,
+                         OptionalAttributesSet & optionalAttributes) :
+            endpointId(endpoint),
+            mFeatureFlags(featureFlags), mEnabledOptionalAttributes(optionalAttributes)
+        {}
 
-    CHIP_ERROR Read(const app::ConcreteReadAttributePath & aPath, app::AttributeValueEncoder & aEncoder) override;
+        EndpointId endpointId;
+        const BitFlags<ElectricalEnergyMeasurement::Feature> mFeatureFlags;
+        OptionalAttributesSet mEnabledOptionalAttributes;
+    };
 
-    bool HasFeature(Feature aFeature) const;
-    bool SupportsOptAttr(OptionalAttributes aOptionalAttrs) const;
+    ElectricalEnergyMeasurementCluster(const Config & config);
+
+    OptionalAttributesSet & OptionalAttributes() { return mEnabledOptionalAttributes; }
+    const OptionalAttributesSet & OptionalAttributes() const { return mEnabledOptionalAttributes; }
+    const BitFlags<ElectricalEnergyMeasurement::Feature> & Features() const { return mFeatureFlags; }
+
+    // Direct access to measurement data - for backwards compatibility
+    const ElectricalEnergyMeasurement::MeasurementData * GetMeasurementData() const { return &mMeasurementData; }
+
+    // Getters - return copies with error checking
+    CHIP_ERROR GetMeasurementAccuracy(MeasurementAccuracyStruct & outValue) const;
+    CHIP_ERROR GetCumulativeEnergyImported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetCumulativeEnergyExported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetPeriodicEnergyImported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetPeriodicEnergyExported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetCumulativeEnergyReset(Optional<CumulativeEnergyResetStruct> & outValue) const;
+
+    // Setters - update values and notify data model
+    CHIP_ERROR SetMeasurementAccuracy(const MeasurementAccuracyStruct & value);
+    CHIP_ERROR SetCumulativeEnergyImported(const Optional<EnergyMeasurementStruct> & value);
+    CHIP_ERROR SetCumulativeEnergyExported(const Optional<EnergyMeasurementStruct> & value);
+    CHIP_ERROR SetPeriodicEnergyImported(const Optional<EnergyMeasurementStruct> & value);
+    CHIP_ERROR SetPeriodicEnergyExported(const Optional<EnergyMeasurementStruct> & value);
```

Comment:
Why are these public?  The public API is CumulativeEnergySnapshot/PeriodicEnergySnapshot which does the right things; these setters do not necessarily do all the right things and should not be used by applications.

Reply:
In the event where the app wants to update the value without generating an event. Do we want the only way to update our attributes to be generating an event?

Reply:
Does the spec not require that the event is generated whenever the attributes change?  If it does (and I recall it does), then we shouldn't be adding public API that just results in spec violations.  If it does not, then this is fine, but then we should document how this differs from the API that _does_ generate the event and explain in the API documentation, with references to the spec, which API should be used in which circumstances.